src/hypervisor.c

```diff
@@ -89,9 +89,15 @@ cc_oci_expand_netdev_cmdline(struct cc_oci_config *config, guint index) {
 /* "pcie.0" is the child pci bus available for device "pci-lite-host".
  * Use a pci slot available on that bus after adding an offset to take 
  * into account busy slots and the slots used earlier in our qemu options.
+ *
+ * Use this as fallback only if mac address is empty for some reason.
+ * Else simply pass the interface with just the mac address without 
+ * assigning the interface to a partcular bus or slot. This allows for more
+ * flexibility.
  */
-#define QEMU_FMT_DEVICE "driver=virtio-net-pci,bus=/pci-lite-host/pcie.0,addr=%x,netdev=%s"
-#define QEMU_FMT_DEVICE_MAC QEMU_FMT_DEVICE ",mac=%s"
+
+#define QEMU_FMT_DEVICE_WITHOUT_MAC "driver=virtio-net-pci,bus=/pci-lite-host/pcie.0,addr=%x,netdev=%s"
+#define QEMU_FMT_DEVICE_MAC "driver=virtio-net-pci,netdev=%s,mac=%s"
```

Comment:
What about something like this to avoid the duplication? I'd also go with `NO_MAC` rather than `WITHOUT_MAC` as it's shorter but still meaningful:
```
#define QEMU_FMT_DEVICE_COMMON "driver=virtio-net-pci,netdev=%s"
#define QEMU_FMT_DEVICE_NOT_MAC QEMU_FMT_DEVICE_COMMON ",bus=/pci-lite-host/pcie.0,addr=%x"
#define QEMU_FMT_DEVICE_MAC QEMU_FMT_DEVICE_COMMON ",mac=%s"
```